types/node/index.d.ts

```diff
@@ -2352,6 +2352,29 @@ declare module "net" {
     import * as stream from "stream";
     import * as events from "events";
 
+    export interface SocketConstructorOpts {
+        fd?: number;
+        allowHalfOpen?: boolean;
+        readable?: boolean;
+        writable?: boolean;
+    }
+
+    export interface TcpSocketConnectOpts {
+        port: number;
+        host?: string;
+        localAddress?: string;
+        localPort?: number;
+        hints?: number;
+        family?: number;
+        lookup?: Function;
```

Comment:
Would be nice of the lookup property could be typed more concrete. I think it should be of same type as `dns.lookup()`, the variant using `LookupOneOptions`.